enterprise/templates/enterprise/grant_data_sharing_permissions.html

```diff
@@ -1,73 +1,88 @@
-                </form>
-            </div>
-        </section>
-    </div>
-</main>
+  <%static:css group='style-vendor'/>
+    <%! main_css = "style-main-v2" %>
+<%static:css group='${self.attr.main_css}'/>
+  <%!
+    from django.utils.translation import ugettext as _
+    %>
+<html>
+  <title>${ title_bar_prefix } | ${ platform_name }</title>
+  <head>
+    <link rel="stylesheet" href="${static.url('css/vendor/font-awesome.css')}"/>
+    <link rel="stylesheet" href="${static.url('enterprise/grant_data_sharing_permissions.css')}"/>
+    <meta name="viewport" content="user-scalable = no, width=device-width, initial-scale=1"/>
+    <script type="text/javascript" src="${static.url('common/js/vendor/jquery.js')}"></script>
+    <script type="text/javascript" src="${static.url('js/vendor/jquery.cookie.js')}"></script>
+    <script type="text/javascript" src="${static.url('enterprise/grant_data_sharing_permissions.js')}"></script>
+  </head>
+  <body>
+    <header class="logo-container" aria-label="${ title_bar_prefix }">
+      <h1>
+        <img src="${static.url('images/logo.png')}" alt="${ platform_name }"/>
+      </h1>
+    </header>
+    <hr class="pinstripe"/>
+    <main>
+      <div class="consent-container">
+        <h2 class="consent-title">${ consent_message_header }</h2>
+        <div class="consent-message">
+          <p>${ consent_request_prompt }</p>
+          <p>${ requested_permissions_header }
+          <ul class="consent-items">
+            % for permission in requested_permissions:
+            <li>${ permission }</li>
+            % endfor
+          </ul>
+          </p>
+        </div>
+        <div class="consent-input-container login-register">
+          <p class="agreement-text">${agreement_text}</p>
+          <form name="data-sharing-consent" method="POST" id="data-sharing">
+            <button type="submit" class="background-input consent-agreement-button" id="consent-button" name="data_sharing_consent" value="True">${ continue_text }</button>
+            <button type="button" class="failure-link background-input" id="failure-link">${ abort_text }</a>
+            <input type="hidden" name="csrfmiddlewaretoken" value="${ csrf_token }" />
+            % if course_specific and course_id:
+            <input type="hidden" name="course_id" value="${ course_id }" />
+            % endif
+            % if redirect_url:
+            <input type="hidden" name="redirect_url" value="${redirect_url}" />
+            % endif
+            % if failure_url:
+            <input type="hidden" name="failure_url" value="${failure_url}" />
+            % endif
+            % if enrollment_deferred:
+            <input type="hidden" name="enrollment_deferred" value="true" />
+            % endif
+          </form>
+        </div>
+        <br/>
+        <button type="button" id="consent-policy-dropdown-bar" class="consent-policy-dropdown-bar background-input" aria-controls="consent-policy">
```

Comment:
I'm going to argue that this disclosure should appear before the consent buttons in DOM (and visual) order.  As a non-visual user, they are being asked to agree to terms they have not been introduced to yet.  They would have no indication that the details of this agreement are on the page since they follow the confirmation in the reading order.